Sources/Jobs/JobQueueHandler.swift

```diff
@@ -112,20 +114,36 @@ final class JobQueueHandler<Queue: JobQueueDriver>: Sendable {
                     self.updateJobMetrics(for: job.name, startTime: startTime, error: error)
                     return
                 } catch {
-                    if count <= 0 {
-                        logger.debug("Job failed")
+                    if job.didFail {
+                        logger.debug("Job: failed")
                         try await self.queue.failed(jobId: queuedJob.id, error: error)
                         self.updateJobMetrics(for: job.name, startTime: startTime, error: error)
                         return
                     }
-                    count -= 1
-                    logger.debug("Retrying Job")
-                    self.updateJobMetrics(for: job.name, startTime: startTime, retrying: true)
+
+                    let attempts = job.attempts + 1
+
+                    let delay = self.backoff(attempts: attempts)
+
+                    try await self.queue.retry(
```

Comment:
I thought of the retry as updating the current state of a job instead of appending a job to the queue. The Postgres implementation would update the job buffer. Now, that I think of it it could insert and oncmlict do update instead.

Reply:
I thought of this a bit more. If the retry func isn't welcome, the push function will need updated to take in a job ID as most persisted drivers will need this id to update the delayed until parameters and buffer. Moreover, if we don't want to embed `attempts` in the job buffer, we would need a solution for the Redis driver. The Postgres driver would just need one more column.

Reply:
The way I saw it was we are creating a new job. eg
```swift
self.queue.finish(jobId)
self.queue.push(jobData, options: .init(delayUntil: delay))
```
Replacing retry() with the above means we still pass all the tests

Reply:
We may need an `updateJob` function at some point though. But I don't think we need it here.

Reply:
I see! For some reason, I didn't see this message until now.

Reply:
This has been addressed!